File: LeetCode/python/1.two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        length = len(nums)
        
        for i in range(length):
            diff = target - nums[i]
            if diff in hashMap:
                return [i, hashMap[diff]]
            hashMap[nums[i]] = i

# Time Complexity: O(n)
# Space Complexity: O(n)

File: LeetCode/php/1. Two Sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $map = array();
        foreach ($nums as $key=>$num){
            $map[$num]=$key;
        }
        foreach($nums as $key => $value){
            $targetItem = $target-$value;
            if(isset($map[$targetItem]) && $key!=$map[$targetItem]){
                return array($key,$map[$targetItem]);
            }
        }
        return array(0,0);
    }
}

Input: nums = [2,7,11,15], target = 9
Output: [0,1]
